frontends/web/src/routes/settings/manage-accounts.tsx

```diff
@@ -196,139 +134,177 @@ class ManageAccounts extends Component<Props, State> {
-                        <p>{t('newSettings.appearance.enableAccount.description')}</p>
-                        <DialogButtons>
-                          <Button
-                            disabled={!currentlyEditedAccount.name}
-                            primary
-                            type="submit">
-                            {t('button.update')}
-                          </Button>
-                        </DialogButtons>
-                      </form>
-                    </Dialog>
-                  )}
-                </WithSettingsTabs>
-              </ViewContent>
-            </View>
-          </Main>
-        </GuidedContent>
-        <AccountGuide accounts={accounts}/>
-      </GuideWrapper>
-    );
-  }
-}
+  const renderAccounts = (accounts: accountAPI.IAccount[]) => {
+    return accounts.filter(account => !account.isToken).map(account => {
+      const active = account.active;
+      const tokensVisible = showTokens[account.code];
+      return (
+        <div key={account.code} className={style.setting}>
+          <div
+            className={`${style.acccountLink} ${active ? style.accountActive : ''}`}
+            onClick={() => active && route(`/account/${account.code}`)}>
+            <Logo stacked active={account.active} className={`${style.coinLogo} m-right-half`} coinCode={account.coinCode} alt={account.coinUnit} />
+            <span className={!account.active ? style.accountNameInactive : ''}>
+              {account.name}
+              {' '}
+              <span className="unit">({account.coinUnit})</span>
+            </span>
+          </div>
+          <div className="flex flex-items-center">
+            {!account.active ? <p className={`text-small ${style.disabledText}`}>{t('generic.enabled_false')}</p> : null}
+            <Button
+              className={style.editBtn}
+              onClick={() => setCurrentlyEditedAccount(account)}
+              transparent>
+              <EditActive />
+              <span className="hide-on-small">
+                {t('manageAccounts.editAccount')}
+              </span>
+            </Button>
+          </div>
+          {active && account.coinCode === 'eth' ? (
+            <div className={style.tokenSection}>
+              <div className={`${style.tokenContainer} ${tokensVisible ? style.tokenContainerOpen : ''}`}>
+                {renderTokens(account.code, account.activeTokens)}
+              </div>
+              <Button
+                className={`${style.expandBtn} ${tokensVisible ? style.expandBtnOpen : ''}`}
+                onClick={() => toggleShowTokens(account.code)}
+                transparent>
+                {t(tokensVisible ? 'manageAccounts.settings.hideTokens' : 'manageAccounts.settings.showTokens', {
+                  activeTokenCount: `${account.activeTokens?.length || 0}`
+                })}
+              </Button>
+            </div>
+          ) : null}
+        </div>
+      );
+    });
+  };
```

Comment:
In the future I think it could make sense to move renderAccounts and renderTokens into their own components within this file.